BasedHardware/omi
# Sdcard Support #570

#570 

How to test:

1. Flash the uf2 file in the file storage_build. Hopefully you have the sd card pins configured as in the hardware schematic. Please make sure your sd card is already in the slot when it turns on, otherwise it will not work.

2. To ensure this works, sever the connection from the friend app. After the flashing sequence the device should turn red, but it should start recording to the sd card. 

3. Use the python file "discover_devices.py" to find your device id. Then in the file "get_info_list.py", place your id on line 12, labeled device_id. Then run the file. The expected result is a list of numbers. Numbers from left to right are the oldest created files to the newest created files. The amount of numbers is the amount of files. For examples, one list of numbers could be

[1000 10000 20000]

where 1000 is the oldest file with 1000 bytes, 10000 is the second file with 10000 bytes, and 20000 is the newest file with 20000 bytes. 

5.On line 13 of the file "get_audio_file.py", place your device id. The run the file. You should be presented with a large wall of numbers indicating opus packets. This file downloads all available files on the device. Each packet is 103 bytes long. The first three bytes are the same as if mic data was sent. The fourth byte is how many numbers are in the packet. Let's call it x. The next x bytes are the opus-encoded numbers. The rest of the packet is filled with zeroes. All of this is stored into the file "my_file.txt"

6. Finally, "decode_audio.py" attempts to decode the file. My opus files are corrupted so I can't verify, but each packet at least appears to be preserved. You are welcome to try.

For now, you can open the serial port to see a wall of text showing the data is being streamed into the sd card. 

If you want to create another file on the sd card, simply reset the device and new data will pour into a separate file. You can always check with "get_info_list.py."

The mic should work as normal if you connect it to the friend app. Any blue tooth connection stops writes to the sd card. Any disconnection should resume data transfer to the sd card. You can always check with get_info_list.
